pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarClientUtils.java

```diff
@@ -62,6 +62,7 @@ public static ClientConfigurationData newClientConf(String serviceUrl, Propertie
             clientConf.setAuthParams(properties.getProperty(PulsarOptions.AUTH_PARAMS_KEY));
             clientConf.setAuthPluginClassName(
                     properties.getProperty(PulsarOptions.AUTH_PLUGIN_CLASSNAME_KEY));
+            clientConf.setTlsTrustCertsFilePath(PulsarOptions.TLS_TRUSTCERTS_FILEPATH);
```

Comment:
The Pulsar client cannot support JWT authentication with TLS transport, which only provides the use of "AuthenticationTls" to set up TLS transport/authentication, this is a conflict with JWT authentication.

BTW, the trust certificate is a CA certificate, this is useless. Usually, we use mTLS, which requires a certificate and private key.

Reply:
Previously, I submitted https://github.com/apache/pulsar/pull/15634 to improve this, welcome to review if you are interesting.

Reply:
@nodece  Yes,  Our cluster scene is JWT for authentication and TLS for transport encryption. 
And, tls requires a ca.cert.pem file.

Reply:
Yes, if you do not use mTLS, this PR is right ok.

Reply:
Could you add a test for these changes?